### PR TITLE
etcdmain: start on unsupported arch when ETCD_UNSUPPORTED_ARCH is set

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,8 +141,9 @@ The `v2` API responses should not change after the 2.0.0 release but new feature
 
 etcd has known issues on 32-bit systems due to a bug in the Go runtime. See #[358][358] for more information.
 
-To avoid inadvertantly producing an unstable etcd server, 32-bit builds emit an `etcd` that prints
-a warning message and immediately exits.
+To avoid inadvertantly running a possibly unstable etcd server, `etcd` on unsupported architectures will print
```
Comment (xiang90) [inline README.md]: shall we mention that the only supported arch is amd64 right now?

File: etcdmain/etcd.go [hunk 2]
```diff
@@ -79,6 +76,8 @@ var (
 )
 
 func Main() {
+	checkSupport()
```
Comment (xiang90) [inline etcdmain/etcd.go]: checkSupportArch?

File: etcdmain/etcd.go [hunk 3]
```diff
@@ -605,3 +604,15 @@ func setupLogging(cfg *config) {
...
 	}
 }
+
+func checkSupport() {
+	// TODO qualify arm64
+	if runtime.GOARCH == "amd64" {
+		return
+	}
+	if env, ok := os.LookupEnv("ETCD_UNSUPPORTED_ARCH"); ok && env == runtime.GOARCH {
+		return
```
Comment (xiang90) [inline etcdmain/etcd.go]: print out warning here? like etcd is running with unsupported arch X since unsupported_arch env is set?

**Comment by heyitsanthony** [comment]
all fixed /cc @xiang90

**Comment by xiang90** [comment]
LGTM

**Pull request merged**